app/assets/javascripts/dashboard/views/dataset/team_assignment_modal.js

```diff
@@ -0,0 +1,103 @@
+// @flow
+import _ from "lodash";
+import * as React from "react";
+import { Modal, Spin, Select } from "antd";
+import Request from "libs/request";
+import type { APITeamType } from "admin/api_flow_types";
+import type { DatasetType } from "dashboard/views/dataset_view";
+
+const { Option } = Select;
+
+type Props = {
+  isVisible: boolean,
+  onOk: Function,
+  onCancel: Function,
+  dataset: DatasetType,
+};
+
+type State = {
+  teams: Array<APITeamType>,
+  isLoading: boolean,
+  selectedTeams: Array<string>,
+};
+
+class TeamAssignmentModal extends React.PureComponent<Props, State> {
+  state = {
+    teams: [],
+    isLoading: true,
+    selectedTeams: [],
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillReceiveProps(newProps: Props) {
+    this.setState({
+      selectedTeams: newProps.dataset ? newProps.dataset.allowedTeams : [],
```

Comment:
Isn't `dataset` a non-optional prop? Why do you have to check whether it's defined? Also, note the following "React doesn't call componentWillReceiveProps with initial props during mounting". This means, on mounting the allowedTeams of the given dataset prop is ignored?

Also:
"Note that React may call this method even if the props have not changed, so make sure to compare the current and next values if you only want to handle changes. This may occur when the parent component causes your component to re-render."

That could lead to weird behavior (the selection could get resetted).

Maybe it's easier to not mount the modal at all, if it shouldn't be visible (i.e., a conditional render in the parent component). Then, you can simply initialize the selectedTeams in `componentDidMount`.